3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -118,31 +70,28 @@ function add_member() {
   local replset_addr
   replset_addr="${MONGODB_REPLICA_NAME}/${endpoints}"
 
-  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "${script}" --quiet; then
+  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "while (!rs.add('${host}').ok) { sleep(100); }" --quiet; then
```

Comment:
I think there is a difference.  My intention is that the mongo rs config, once set, stays the same (unless the user changes the number of replicas).  The rs config stores the DNS names, so that if a pod dies and is rescheduled (and its IP address potentially changes in the process), the rs config need not be updated.  Mongo re-resolves the DNS name at each new connection attempt, so the potential IP address change is transparent to mongo.